autoarray/plot/visuals/two_d.py

```diff
@@ -0,0 +1,89 @@
+from matplotlib import patches as ptch
+from typing import List, Optional, Union
+
+from autoarray.mask.mask_2d import Mask2D
+from autoarray.plot.visuals.abstract import AbstractVisuals
+from autoarray.structures.arrays.uniform_1d import Array1D
+from autoarray.structures.arrays.uniform_2d import Array2D
+from autoarray.structures.grids.uniform_2d import Grid2D
+from autoarray.structures.grids.irregular_2d import Grid2DIrregular
+from autoarray.structures.vectors.irregular import VectorYX2DIrregular
+
+
+class Visuals2D(AbstractVisuals):
+    def __init__(
+        self,
+        origin: Optional[Grid2D] = None,
+        mask: Optional[Mask2D] = None,
+        border: Optional[Grid2D] = None,
+        lines: Optional[List[Array1D]] = None,
+        positions: Optional[Union[Grid2DIrregular, List[Grid2DIrregular]]] = None,
+        grid: Optional[Grid2D] = None,
+        mesh_grid: Optional[Grid2D] = None,
+        vectors: Optional[VectorYX2DIrregular] = None,
+        patches: Optional[List[ptch.Patch]] = None,
+        array_overlay: Optional[Array2D] = None,
+        parallel_overscan=None,
+        serial_prescan=None,
+        serial_overscan=None,
+        indexes=None,
+        pix_indexes=None,
+    ):
+
+        self.origin = origin
+        self.mask = mask
+        self.border = border
+        self.lines = lines
+        self.positions = positions
+        self.grid = grid
+        self.mesh_grid = mesh_grid
+        self.vectors = vectors
+        self.patches = patches
+        self.array_overlay = array_overlay
+        self.parallel_overscan = parallel_overscan
+        self.serial_prescan = serial_prescan
+        self.serial_overscan = serial_overscan
+        self.indexes = indexes
+        self.pix_indexes = pix_indexes
+
+    def plot_via_plotter(self, plotter, grid_indexes=None, mapper=None):
+
+        if self.origin is not None:

```

Comment:
This might be more elegant if you have a dictionary mapping types to plotters and then iterate through a list of objects getting each plotter